sources/packages/api/src/cra-integration/cra-integration.service.ts

```diff
@@ -182,104 +148,54 @@ export class CRAIntegrationService {
+   * Downloads the file specified on 'remoteFilePath' parameter from the
    * CRA response folder on the SFTP.
-   * @param fileName File to be downloaded.
+   * @param remoteFilePath full remote file path with file name.
    * @returns Parsed records from the file.
    */
-  async downloadResponseFile(fileName: string): Promise<CRAsFtpResponseFile> {
-    const client = await this.getClient();
-    try {
-      const filePath = `${this.craConfig.ftpResponseFolder}/${fileName}`;
-      // Read all the file content and create a buffer.
-      const fileContent = await client.get(filePath);
-      // Convert the file content to an array of text lines and remove possible blank lines.
-      const fileLines = fileContent
-        .toString()
-        .split(/\r\n|\n\r|\n|\r/)
-        .filter((line) => line.length > 0);
-      // Read the first line to check if the header code is the expected one.
-      const header = CRAFileHeader.CreateFromLine(fileLines.shift()); // Read and remove header.
-      if (header.transactionCode !== TransactionCodes.ResponseHeader) {
-        this.logger.error(
-          `The CRA file ${fileName} has an invalid transaction code on header: ${header.transactionCode}`,
-        );
-        // If the header is not the expected one, just ignore the file.
-        return null;
-      }
-
-      // Remove footer (not used).
-      fileLines.pop();
-
-      // Generate the records.
-      let lineNumber = 1;
-      const statusRecords: CRAResponseStatusRecord[] = [];
-      const totalIncomeRecords: CRAResponseTotalIncomeRecord[] = [];
-      for (const line of fileLines) {
-        const craRecord = new CRAResponseRecordIdentification(line, lineNumber);
-        switch (craRecord.transactionSubCode) {
-          case TransactionSubCodes.ResponseStatusRecord:
-            statusRecords.push(new CRAResponseStatusRecord(line, lineNumber));
-            break;
-          case TransactionSubCodes.TotalIncome:
-            totalIncomeRecords.push(
-              new CRAResponseTotalIncomeRecord(line, lineNumber),
-            );
-            break;
-        }
-        lineNumber++;
-      }
-
-      return {
-        fileName,
-        filePath,
-        statusRecords,
-        totalIncomeRecords,
-      };
-    } finally {
-      await SshService.closeQuietly(client);
+  async downloadResponseFile(
+    remoteFilePath: string,
+  ): Promise<CRAsFtpResponseFile> {
+    const fileLines = await this.downloadResponseFileLines(remoteFilePath);
+    // Read the first line to check if the header code is the expected one.
+    const header = CRAFileHeader.CreateFromLine(fileLines.shift()); // Read and remove header.
+    if (header.transactionCode !== TransactionCodes.ResponseHeader) {
+      this.logger.error(
+        `The CRA file ${remoteFilePath} has an invalid transaction code on header: ${header.transactionCode}`,
+      );
+      // If the header is not the expected one, just ignore the file.
+      return null;
```

Comment:
promise is `CRAsFtpResponseFile`

Reply:
I will change to CRAsFtpResponseFile | null

Reply:
Fixed!